main.cpp

```diff
@@ -477,27 +502,40 @@ void FreightTrain::printMessage(std::string name)
                 << this->numberOfFreightCars << " freight cars and its conductors name is " << this->conductorName << std::endl;
 }
 
+struct TrainWrapper
+{
+    TrainWrapper(FreightTrain* ptr) : trainPtr(ptr){}
+    ~TrainWrapper()
+    {
+        delete trainPtr;
+    }
+    FreightTrain* trainPtr = nullptr;
+};
+
 /*
  new UDT 4:
  with 2 member functions
  */
 struct FreightLine
 {
-    FreightTrain train1 { 4, 20, "Fred" };
-    FreightTrain train2 { 2, 25, "Bob" };
-    FreightTrain train3;
+    TrainWrapper train1 = TrainWrapper(new FreightTrain(4,20,"Fred"));
+    TrainWrapper train2 = TrainWrapper(new FreightTrain(2,25,"Bob"));
+    TrainWrapper train3 = TrainWrapper(new FreightTrain());
+    //TrainWrapper train3(new FreightTrain()); // not sure I understand why I can't declare and initialize wrapper class member variables like this
 
     FreightLine();
     ~FreightLine();
 
     bool DispatchTrain(int train, std::vector<int> pickupDropoffList);
-    void DescribeTrain(FreightTrain train);
+    bool DispatchTrain(FreightTrain& train, std::vector<int> pickupDropoffList);
+    void DescribeTrain(const FreightTrain& train);
```

Comment:
member function names should begin with a lower-case first letter.
- [ ] revise